From 617c5e592f15b5662f9c5067a8f25e81d28a6cd2 Mon Sep 17 00:00:00 2001
From: Dan Lambright <dlambrig@redhat.com>
Date: Fri, 23 Oct 2015 12:10:42 -0400
Subject: [PATCH 72/98] cluster/tier enable CTR on attach tier

This is a backport of 12420

CTR is currently disabled by default, and must be manually enabled
for tiering to start. This is an overhead on the administrator and
easy to overlook. Enable it automatically when a tier is attached.

> Change-Id: I0c29de8762faec1bfe6d1376a57eeef3357ad15a
> BUG: 1274847
> Signed-off-by: Dan Lambright <dlambrig@redhat.com>
> Reviewed-on: http://review.gluster.org/12420
> Tested-by: Gluster Build System <jenkins@build.gluster.com>
> Reviewed-by: mohammed rafi  kc <rkavunga@redhat.com>
Signed-off-by: Dan Lambright <dlambrig@redhat.com>

Change-Id: I98cd2f17641049f4770dd9b231dea72bb2e6ceee
BUG: 1276678
Reviewed-on: https://code.engineering.redhat.com/gerrit/60525
Reviewed-by: Dan Lambright <dlambrig@redhat.com>
Tested-by: Dan Lambright <dlambrig@redhat.com>
---
 tests/basic/tier/tier.t                        |    2 --
 xlators/mgmt/glusterd/src/glusterd-brick-ops.c |    3 +++
 2 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/tests/basic/tier/tier.t b/tests/basic/tier/tier.t
index 55f6696..bc5e4ce 100755
--- a/tests/basic/tier/tier.t
+++ b/tests/basic/tier/tier.t
@@ -141,8 +141,6 @@ TEST $CLI volume start $V0
 TEST $CLI volume set $V0 performance.quick-read off
 TEST $CLI volume set $V0 performance.io-cache off
 
-TEST $CLI volume set $V0 features.ctr-enabled on
-
 #Not a tier volume
 TEST ! $CLI volume set $V0 cluster.tier-demote-frequency 4
 
diff --git a/xlators/mgmt/glusterd/src/glusterd-brick-ops.c b/xlators/mgmt/glusterd/src/glusterd-brick-ops.c
index 9d82e14..3ca5dcf 100644
--- a/xlators/mgmt/glusterd/src/glusterd-brick-ops.c
+++ b/xlators/mgmt/glusterd/src/glusterd-brick-ops.c
@@ -2200,6 +2200,9 @@ glusterd_op_perform_attach_tier (dict_t *dict,
         volinfo->tier_info.hot_type      = type;
         ret = dict_set_int32 (dict, "type", GF_CLUSTER_TYPE_TIER);
 
+        if (!ret)
+                ret = dict_set_str (volinfo->dict, "features.ctr-enabled", "on");
+
         return ret;
 }
 
-- 
1.7.1

